Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1

```diff
@@ -0,0 +1,243 @@
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "echo 'chmod 755 /home/$VMUserName/NET-Verify-Boot-NoNIC.sh' >> /etc/rc.d/rc.local" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "echo 'cd /home/$VMUserName/ && ./NET-Verify-Boot-NoNIC.sh > /home/$VMUserName/NET-Verify-Boot-NoNIC.log  &' >> /etc/rc.d/rc.local" -runAsSudo
+    }
+    elseif ($linuxRelease -eq "UBUNTU") {
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "systemctl start cron.service" -runAsSudo
+        RunLinuxCmd -username $VMUserName  -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "echo '@reboot chmod 755 /home/$VMUserName/NET-Verify-Boot-NoNIC.sh' >> /var/spool/cron/crontabs/root"  -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "echo '@reboot cd /home/$VMUserName/ && ./NET-Verify-Boot-NoNIC.sh > /home/$VMUserName/NET-Verify-Boot-NoNIC.log  &' >> /var/spool/cron/crontabs/root"  -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "chmod 0600 /var/spool/cron/crontabs/root" -runAsSudo
+    }
+    elseif ($linuxRelease -eq "SLES") {
+        $cmdToVM = @"
+[Unit]
+ConditionFileIsExecutable=/home/$VMUserName/NET-Verify-Boot-NoNIC.sh
+After=getty.target
+
+[Service]
+Type=idle
+ExecStart= cd /home/$VMUserName/ && ./NET-Verify-Boot-NoNIC.sh
+TimeoutSec=0
+RemainAfterExit=yes
+"@
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "echo '$cmdToVM' > /etc/systemd/system/after-local.service" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "chmod +x /etc/systemd/system/after-local.service" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "chmod 755 /home/$VMUserName/NET-Verify-Boot-NoNIC.sh" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "systemctl daemon-reload && systemctl enable after-local.service" -runAsSudo
+    }
+    else {
+        LogErr "Unknown linux distribution '${linuxRelease}'"
+        return "ABORTED"
+    }
+
+    # Stop the VM
+    LogMsg "Stopping the VM..."
+    Stop-VM -Name "${VMName}" -ComputerName $HvServer -Force -ErrorAction SilentlyContinue
+    if (-not $?) {
+        LogErr "Unable to stop VM to allow removal of original NIC"
+        return "FAIL"
+    }
+
+    # Remove the original NIC
+    LogMsg "Removing the original NIC from the VM..."
+    Remove-VMNetworkAdapter -VMName $VMName -ComputerName $HvServer -ErrorAction SilentlyContinue
+    if (-not $?) {
+        LogErr  "Unable to Hot Remove NIC"
```

Comment:
this is not hot remove, the VM gets stopped above

Reply:
Updated comment to reflect action